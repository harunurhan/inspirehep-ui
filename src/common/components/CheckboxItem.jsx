import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Checkbox } from 'antd';

class CheckboxItem extends Component {
  constructor(props) {
    super(props);
    this.onChange = this.onChange.bind(this);
    // TODO: maybe move to deriveStateFromProps?
    this.state = {
      checked: this.props.checked,
    };
  }

  onChange(event) {
    const { checked } = event.target;
    this.setState({ checked });

    if (this.props.onChange) {
      this.props.onChange(checked);
    }
  }

  render() {
    return (
      <Checkbox
        onChange={this.onChange}
        checked={this.state.checked}
      >
        {this.props.children}
      </Checkbox>
    );
  }
}

CheckboxItem.propTypes = {
  checked: PropTypes.bool,
  children: PropTypes.node.isRequired,
  onChange: PropTypes.func,
};

CheckboxItem.defaultProps = {
  checked: false,
  onChange: null,
};

export default CheckboxItem;
