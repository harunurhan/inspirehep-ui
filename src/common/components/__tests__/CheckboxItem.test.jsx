import React from 'react';
import { shallow } from 'enzyme';

import CheckboxItem from '../CheckboxItem';

describe('CheckboxItem', () => {
  it('render initial state with all props set', () => {
    const wrapper = shallow((
      <CheckboxItem
        onChange={jest.fn()}
        checked
      >
        Test
      </CheckboxItem>
    ));
    expect(wrapper).toMatchSnapshot();
  });

  describe('onChange', () => {
    it('sets state and calls onChange if present', () => {
      const onChange = jest.fn();
      const wrapper = shallow((
        <CheckboxItem
          onChange={onChange}
          checked
        >
          Test
        </CheckboxItem>
      ));
      wrapper.instance().onChange({ target: { checked: true } });
      expect(wrapper.instance().state.checked).toBe(true);
      expect(onChange).toHaveBeenCalledWith(true);
    });

    it('sets state only if present onChange is absent', () => {
      const wrapper = shallow((
        <CheckboxItem
          checked
        >
          Test
        </CheckboxItem>
      ));
      wrapper.instance().onChange({ target: { checked: true } });
      expect(wrapper.instance().state.checked).toBe(true);
    });
  });
});
