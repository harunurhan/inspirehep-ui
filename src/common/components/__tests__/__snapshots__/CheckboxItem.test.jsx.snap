// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxItem render initial state with all props set 1`] = `
<Checkbox
  checked={true}
  indeterminate={false}
  onChange={[Function]}
  prefixCls="ant-checkbox"
>
  Test
</Checkbox>
`;
